File: Model/Movie.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace Model
{
    public class Movie
    {
        public int Id { get; set; }
        public string MovieName { get; set; }
        public string MovieStar { get; set; }
        public string Description { get; set; }
        public DateTime ReleaseDate { get; set; }
        public decimal Price { get; set; }
        public decimal StarRating { get; set; }
        public string ImageUrl { get; set; }
        //public int? MovieEventId { get; set; }
        public IList<Artist>Artists { get; set; }
        public IList<MovieEvent>MovieEvents { get; set; }

    }
}

File: MVCDemo/app/movies/movie.ts
  
export interface IMovie {
     id: number;
     movieName: string;
     movieStar: string;
     description: string;
     releaseDate: Date;
     price: number;
     starRating: number;
     imageUrl: string;
 }

export class Movie {
  constructor(
    id: number,
     movieName: string,
     movieStar: string,
     description: string,
     releaseDate: Date,
     price: number,
     starRating: number,
     imageUrl: string) 
    { }
}

export interface IMovieEvents {
    id?: number;
    EventDescription: string;
    EventDate: Date; 
    Place: string;
    MovieId: number;
}

 